engine/src/main/java/org/openobservatory/engine/Engine.java

```diff
@@ -29,6 +29,11 @@ public static String getVersionMK() {
         return io.ooni.mk.MKVersion.getVersionMK();
     }
 
+    /** getUUID returns the UUID4 generated for this client  */
```

Comment:
I disagree with this documentation and I am not sure now the usage is correct. This function returns a new UUID4 every time you invoke it. The way in which it's documented and the name of the function seem to suggest that such UUID4 is cached somewhere and reused, which is not the case. So,

1. I strongly suggest to rename this function `newUUID4`

2. I suggest we make sure we're using this function correctly

Regarding 2., I will insert a specific comment where needed